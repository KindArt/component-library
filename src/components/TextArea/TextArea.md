Standard:

```tsx padded
<TextArea label="Label" placeholder="Placeholder" />
```

Disabled:

```tsx padded
<TextArea label="Label" placeholder="Placeholder" disabled />
```

Error State:

```tsx padded
<TextArea label="Label" errorMessage="This is an error" />
```
