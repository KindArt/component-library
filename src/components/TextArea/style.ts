import { HTMLProps } from 'react';
import styled, { css } from 'styled-components';
import { Colours, CssVariables } from '../../constants/styles';
import { ITextAreaProps } from '.';

export const TextAreaWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

interface ITextAreaField {
  errorMessage?: string;
  ref: any;
}

export const TextAreaField = styled.textarea<ITextAreaField>`
  border: ${({ theme }) => theme.textArea.border};
  border-radius: 4px;
  padding: 10px 15px;
  font-size: 0.875rem;
  resize: none;

  &:disabled {
    background: rgba(62, 63, 66, 0.3);
    cursor: not-allowed;
  }

  &:focus {
    border-color: ${({ theme }) => theme.textArea.focusColour};
  }

  ${({ errorMessage, theme }) =>
    errorMessage &&
    css`
      border-color: ${theme.textArea.errorBorderColor};
    `}

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

TextAreaField.defaultProps = {
  theme: {
    textArea: {
      border: `${CssVariables.BORDER_DEFAULT}`,
      errorBorderColor: `${Colours.ERROR}`,
      focusColour: `${Colours.PRIMARY}`,
    },
  },
};

export const ErrorMessage = styled.p`
  color: ${({ theme }) => theme.errorMessage.color};
  margin-top: 6px;
  margin-bottom: 0;
  line-height: 1.375rem;

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

ErrorMessage.defaultProps = {
  theme: {
    errorMessage: {
      color: `${Colours.ERROR}`,
    },
  },
};
