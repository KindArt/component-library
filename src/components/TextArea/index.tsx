import React, { forwardRef, HTMLProps } from 'react';
import { Label } from '../';
import { TextAreaWrapper, TextAreaField, ErrorMessage } from './style';

export interface ITextAreaProps extends HTMLProps<HTMLTextAreaElement> {
  label?: string;
  errorMessage?: string;
  testIds?: {
    wrapper?: string;
    label?: string;
    textArea?: string;
    errorMessage?: string;
  };
}

const componentName = 'TextArea';

export const TextArea = forwardRef<HTMLTextAreaElement, ITextAreaProps>(
  ({ label, errorMessage, testIds, ...props }, ref) => {
    // const;
    return (
      <TextAreaWrapper>
        {label && <Label content={label} />}
        <TextAreaField ref={ref} errorMessage={errorMessage} {...(props as any)} />
        {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
      </TextAreaWrapper>
    );
  }
);

export default TextArea;
