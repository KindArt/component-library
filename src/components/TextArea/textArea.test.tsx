import React from 'react';
import { render } from '@testing-library/react';
import TextArea from '.';

describe('<TextArea />', () => {
  const getTextArea = (props?: any) => {
    const utils = render(<TextArea {...props} />);
    return { ...utils };
  };

  it('should render', () => {
    const { container } = getTextArea();
    expect(container).toMatchSnapshot();
  });
});
