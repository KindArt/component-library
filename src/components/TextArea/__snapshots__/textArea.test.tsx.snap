// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> should render 1`] = `
<div>
  <div
    class="sc-fzoKki kwfRYe"
  >
    <textarea
      class="sc-fzoYkl hpuNKy"
    />
  </div>
</div>
`;
